narwhals/_dask/group_by.py

```diff
@@ -0,0 +1,203 @@
+
+POLARS_TO_PANDAS_AGGREGATIONS = {
+    "len": "size",
+}
+
+
+class DaskLazyGroupBy:
+    def __init__(self, df: DaskLazyFrame, keys: list[str]) -> None:
+        self._df = df
+        self._keys = keys
+        self._grouped = self._df._native_dataframe.groupby(
+            list(self._keys),
+            dropna=False,
+        )
+
+    def agg(
+        self,
+        *aggs: IntoDaskExpr,
+        **named_aggs: IntoDaskExpr,
+    ) -> DaskLazyFrame:
+        exprs = parse_into_exprs(
+            *aggs,
+            namespace=self._df.__narwhals_namespace__(),
+            **named_aggs,
+        )
+        implementation: Implementation = Implementation.PANDAS
+        output_names: list[str] = copy(self._keys)
+        for expr in exprs:
+            if expr._output_names is None:
+                msg = (
+                    "Anonymous expressions are not supported in group_by.agg.\n"
+                    "Instead of `nw.all()`, try using a named expression, such as "
+                    "`nw.col('a', 'b')`\n"
+                )
+                raise ValueError(msg)
+            output_names.extend(expr._output_names)
+
+        return agg_dask(
+            self._grouped,
+            exprs,
+            self._keys,
+            output_names,
+            self._from_native_dataframe,
+            dataframe_is_empty=False,
+            implementation=implementation,
+            backend_version=self._df._backend_version,
+        )
+
+    def _from_native_dataframe(self, df: DaskLazyFrame) -> DaskLazyFrame:
+        from narwhals._dask.dataframe import DaskLazyFrame
+
+        return DaskLazyFrame(
+            df,
+            backend_version=self._df._backend_version,
+        )
+
+    def __iter__(self) -> Iterator[tuple[Any, DaskLazyFrame]]:
+        with warnings.catch_warnings():
+            # we already use `tupleify` above, so we're already opting in to
+            # the new behaviour
+            warnings.filterwarnings(
+                "ignore",
+                message="In a future version of pandas, a length 1 tuple will be returned",
+                category=FutureWarning,
+            )
+            iterator = self._grouped.__iter__()
+        yield from (
+            (key, self._from_native_dataframe(sub_df)) for (key, sub_df) in iterator
+        )
+
+
+def agg_dask(
+    grouped: Any,
+    exprs: list[DaskExpr],
+    keys: list[str],
+    output_names: list[str],
+    from_dataframe: Callable[[Any], DaskLazyFrame],
+    *,
+    implementation: Any,
+    backend_version: tuple[int, ...],
+    dataframe_is_empty: bool,
+) -> DaskLazyFrame:
+    """
+    This should be the fastpath, but cuDF is too far behind to use it.
+
+    - https://github.com/rapidsai/cudf/issues/15118
+    - https://github.com/rapidsai/cudf/issues/15084
+    """
+    all_simple_aggs = True
+    for expr in exprs:
+        if not is_simple_aggregation(expr):
+            all_simple_aggs = False
+            break
+
+    if all_simple_aggs:
```

Comment:
maybe this is the only path we need to support? check what we do for pyarrow, where we raise if non-simple aggregations are found